src/Client.php

```diff
@@ -58,7 +58,7 @@ class Client
      *
      * @var \Bugsnag\Pipeline
      */
-    protected $pipeline;
+    public $pipeline;
```

Comment:
Can we make this readonly so you cant set the pipeline variable to a string or a totally different array?

Reply:
There's no easy way to make something readonly in PHP it seems, see [here](https://stackoverflow.com/questions/2343790/how-to-implement-a-read-only-member-variable-in-php).

We could use the `__get` method override as described above, but as far as I can tell we'd then have to create entries for each class member, to ensure everything else gets passed back as required.

We could also use a getter method, which isn't nice, but would minimise the required changes and complexity.

Reply:
A getter method would be preferable, as pipeline should not be arbitrarily writeable. The same rationale applies to other Client properties like `$config`, which has `getConfig()`.